### PR TITLE
[dem] Test which compares unbonded spheres and bonded spheres in a case where they should behave equally

### PR DESCRIPTION
**Pull request by salvalatorre**
Adding a comparison between the results of super-weakly bonded pair of spheres against the same pair without the bond. They should behave very similarly. 
The comparison is repeated in the same test for different continuum constitutive laws: DEM_KDEM, DEM_KDEM_with_damage, DEM_KDEM_with_damage_parallel_bond, DEM_KDEM_with_damage_parallel_bond_Hertz.

### CODE REVIEW

File: applications/DEMApplication/python_scripts/DEM_material_test_script.py [hunk 1]
```diff
@@ -199,6 +202,7 @@ def CylinderSkinDetermination(self): #model_part, solver, DEM_parameters):
 
         h = self.height
         d = self.diameter
+        ymin = -0.000195748 #0.0
```
Comment (maceligueta) [inline applications/DEMApplication/python_scripts/DEM_material_test_script.py]: hardcoded number?

File: applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp [hunk 2]
```diff
@@ -123,7 +123,7 @@ namespace Kratos {
             RebuildListOfSphericParticles<SphericParticle>(r_model_part.GetCommunicator().GhostMesh().Elements(), mListOfGhostSphericParticles);
 
             // Search Neighbours and related operations
-            SetSearchRadiiOnAllParticles(*mpDem_model_part, mpDem_model_part->GetProcessInfo()[SEARCH_RADIUS_INCREMENT], 1.0);
+            SetSearchRadiiOnAllParticles(*mpDem_model_part, mpDem_model_part->GetProcessInfo()[SEARCH_RADIUS_INCREMENT_FOR_BONDS_CREATION], 1.0);
```
Comment (maceligueta) [inline applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp:126]: Maybe we could do the same change inside the function SetCoordinationNumber. That function makes searches iteratively.

**Review by maceligueta — CHANGES_REQUESTED** [review]

File: applications/DEMApplication/python_scripts/DEM_material_test_script.py (hunk 1 above)
Comment (salvalatorre) [reply applications/DEMApplication/python_scripts/DEM_material_test_script.py]: Yes, I was thinking about adding those values, ymin and zmin, through the ProjectParams.json. What do you think?

File: applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp (hunk 2 above)
Comment (salvalatorre) [reply applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp:126]: Done!

File: applications/DEMApplication/python_scripts/DEM_material_test_script.py (hunk 1 above)
Comment (salvalatorre) [reply applications/DEMApplication/python_scripts/DEM_material_test_script.py]: Done!

File: applications/DEMApplication/tests/test_DEM_3D_continuum_vs_discontinuum.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+def GetFilePath(fileName):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), fileName)
+
+class DEM3D_ContinuumTestVsDiscontinuumSolution(KratosMultiphysics.DEMApplication.DEM_analysis_stage.DEMAnalysisStage, KratosUnittest.TestCase):
+
+    def InitializeSolutionStep(self):
+        super().InitializeSolutionStep()
+
+        self.tolerance = 2.5e-5
```
Comment (maceligueta) [inline applications/DEMApplication/tests/test_DEM_3D_continuum_vs_discontinuum.py]: can this tolerance be decreased? I was hoping the error would be smaller

Comment (salvalatorre) [reply applications/DEMApplication/tests/test_DEM_3D_continuum_vs_discontinuum.py]: OK, I'll recheck!

**Review by maceligueta — DISMISSED** [review]

**Comment by farrufat-cimne** [comment]
There have been some updates on the KDEM law t(Kn and Kt) that cause a slight different behaviour in some of the tests.
The reference values have been updated.

**Comment by maceligueta** [comment]
> There have been some updates on the KDEM law t(Kn and Kt) that cause a slight different behaviour in some of the tests.
> The reference values have been updated.

@farrufat-cimne can you elaborate a bit more? What were those changes in kn and kt?

**Review by maceligueta — APPROVED** [review]

**Pull request merged**